src/rgw/rgw_sync_module_es.cc

```diff
@@ -801,6 +817,25 @@ class RGWElasticDataSyncModule : public RGWDataSyncModule {
 
   void init(RGWDataSyncEnv *sync_env, uint64_t instance_id) override {
     conf->init_instance(sync_env->store->svc()->zone->get_realm(), instance_id);
+    // try to get elastic search version
+    RGWCoroutinesManager crs(sync_env->store->ctx(), sync_env->store->getRados()->get_cr_registry());
+    RGWHTTPManager http_manager(sync_env->store->ctx(), crs.get_completion_mgr());
+    int ret = http_manager.start();
+    if (ret < 0) {
+      return;
+    }
+    ret = crs.run(new RGWReadRESTResourceCR<ESInfo>(sync_env->cct,
```

Comment:
I see that this init() function is being caled by RGWDataSyncCR::operate(), so it shouldn't make any blocking calls (ie RGWCoroutinesManager::run()) or that will tie up the entire data sync thread.

We should either change the RGWDataSyncModule::init() interface to return a coroutine (similar to init_sync() and start_sync()), or move this RGWReadRESTResourceCR call into the coroutine returned by RGWElasticDataSyncModule::init_sync().

Reply:
see https://github.com/ceph/ceph/pull/32269, we could re-use `start_sync` to solve this issue.